includes/amp-helper-functions.php

```diff
@@ -18,7 +18,7 @@ function amp_get_permalink( $post_id ) {
 
 function post_supports_amp( $post ) {
 	// Because `add_rewrite_endpoint` doesn't let us target specific post_types :(
-	if ( ! post_type_supports( $post->post_type, AMP_QUERY_VAR ) ) {
+	if (!AMP_CANONICAL && !post_type_supports( $post->post_type, AMP_QUERY_VAR ) ) {
```

Comment:
`AMP_CANONICAL` was removed. Might be nice to have a helper function since we do the same checks a few places.